actions/actions.py

```diff
@@ -197,6 +197,16 @@ def validate_location(
         tracker: Tracker,
         domain: Dict[Text, Any],
     ) -> Dict[Text, Optional[Text]]:
+        metadata = tracker.latest_message.get("metadata")
+
+        # Use location pin data if submitted
+        if metadata and metadata.get("type") == "location":
+            return {
+                "location": tracker.metadata["location"]["address"],
```

Comment:
It doesn't look like `address` is a required field, so we should handle that: https://developers.facebook.com/docs/whatsapp/api/messages/others#locations